browser/extensions/extensibles/content/TabFeatures.jsm

```diff
@@ -172,9 +172,26 @@ const TabFeatures = {
   },
 
   _attemptRestart() {
+    // Purge cache if required
     if (Services.prefs.getBoolPref(this.PREF_PURGECACHE)) {
       Services.appinfo.invalidateCachesOnRestart();
     }
+
+    // Notify all windows that an application quit has been requested.
+    let cancelQuit = Cc["@mozilla.org/supports-PRBool;1"].createInstance(
+      Ci.nsISupportsPRBool
+    );
+    Services.obs.notifyObservers(
+      cancelQuit,
+      "quit-application-requested",
+      "restart"
+    );
+
+    // Something aborted the quit process.
+    if (cancelQuit.data) {
```

Comment:
I dunno, but this seems a little "backward" to me...would it make more sense to `if (!cancelQuit.data) { Services.... }` since it's about to return anyway...or no? (I'm a bit archaic on the whole 'return early' practice).

Reply:
The whole block from 179-194 shouldn't actually be in here, so I will remove it. The `quit-application-requested` observer essentially pops up a confirmation box if certain prefs/conditions are met, this blocks execution so that `if (cancelQuit.data) ` isn't evaluated until the user selects a value from the confirmation box (if it appears). What it ultimately triggers is this method: https://searchfox.org/mozilla-esr91/source/browser/components/BrowserGlue.jsm#2725

For this case we have a separate preference entirely controlling whether a confirmation prompt dispays or not, and display that prompt in the `restartBrowser` method, so adding the `notifyObservers` call is redundant.